x-pack/docs/en/security/securing-communications/security-basic-setup-https.asciidoc

```diff
@@ -0,0 +1,595 @@
+===== Sign your own certificates
+
+If you want to use the CA that you created when
+<<generate-certificates,Generating the certificate authority>> answer `n` to
+the first question. You then specify the location of your CA, which the tool
+uses to sign and generate a `.p12` certificate. The steps in this procedure
+follow this workflow.
+
+[discrete]
+===== Sign certificates with a central CA
+
+If you work in an environment with a central security team, they can likely
+generate a certificate for you. Infrastructure within your organization
+might already be configured to trust an existing CA, so it may be easier
+for clients to connect to {es} if you use a CSR and send that
+request to the team that controls your CA. To use a central CA, answer `y` to
+the first question.
+****
+
+[[basic-setup-https-prerequisites]]
+==== Prerequisites
+
+Complete all steps in <<security-basic-setup,Set up basic security for the Elastic Stack>>.
+
+[[encrypt-http-communication]]
+==== Encrypt HTTP client communications for {es}
+
+. Stop {es} and {kib} if they are running.
+
+. From the directory where you installed {es}, run the {es}
+   HTTP certificate tool to generate a Certificate Signing Request (CSR).
++
+[source,shell]
+----
+./bin/elasticsearch-certutil http
+----
++
+This command generates a `.zip` file that contains certificates and keys
+to use with {es} and {kib}. Each folder contains a `README.txt`
+explaining how to use these files.
+
+   a. When asked if you want to generate a CSR, enter `n`.
+
+   b. When asked if you want to use an existing CA, enter `y`.
+
+   c. Enter the path to your CA. This is the absolute path to
+   the `elastic-stack-ca.p12` file that you generated for your cluster.
+
+   d. Enter the password for your CA.
+
+   e. Enter an expiration value for your certificate. You can enter the
+   validity period in years, months, or days. For example, enter `90D` for 90
+   days.
+
+   f. When asked if you want to generate one certificate per node, enter `y`.
++
+Each certificate will have its own private key, and will be issued for a
+specific hostname or IP address.
+
+   g. When prompted, enter the name of the first node in your cluster. Use the same node name from previous steps.
+
+   h. Enter all hostnames used to connect to your first node. These hostnames
+   will be added as DNS names in the Subject Alternative Name (SAN) field in your certificate.
++
+List every hostname and variant used to connect to your cluster over HTTPS.
+
+   i. Enter the IP addresses that clients can use to connect to your node.
+
+   j. Repeat these steps for each additional node in your cluster.
+
+. After generating a certificate for each of your nodes, enter a password for
+   your private key when prompted.
+
+. Unzip the generated `elasticsearch-ssl-http.zip` file. This compressed file
+   contains one directory for both {es} and {kib}.
++
+--
+[source,txt]
+----
+/elasticsearch
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
```

Comment:
This directory layout is when we enter `n` for the question of "do you want to generate one certificate per node". But the previous step `f` says enter `y` to this question.